src/pages/index.js

```diff
@@ -30,36 +33,98 @@ import {
   config,
 } from "../utils/constants.js";
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-section.renderItems();
+const handleAddCardSubmit = (inputValues) => {
+  api.createCard(inputValues).then((res) => {
+    const cardElement = createCard(res);
+    section.addItem(cardElement);
+    newCardPopup.close();
+    addFormValidator.resetValidation();
+  });
+};
 const newCardPopup = new PopupWithForm("#profile-add-modal", (inputValues) => {
-  const cardElement = createCard(inputValues);
-  // cardsList.prepend(cardElement);
-  section.addItem(cardElement);
-  newCardPopup.close();
-  // addForm.reset();
-  addFormValidator.resetValidation();
+  newCardPopup.setSubmitButtonText("Saving....");
+  api
+    .createCard(inputValues)
+    .then((res) => {
+      const cardElement = createCard(res);
+      section.addItem(cardElement);
+      newCardPopup.close();
+      addFormValidator.resetValidation();
+    })
```

Comment:
Needs correction: don't forget to handle the api errors (add a `catch` block for every request)